### PR TITLE
Added jsdoc and grunt-jsdoc

### PR DESCRIPTION
**Pull request by smbuthia**
# Description

Added jsdoc and grunt-jsdoc for automatic code documentation.

medic/medic#5548

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -1042,4 +1050,10 @@ module.exports = function(grunt) {
   grunt.registerTask('default', 'Build and deploy the webapp for dev', [
     'dev-webapp',
   ]);
+
+  grunt.loadNpmTasks('grunt-jsdoc');
```
Comment (garethbowen) [inline Gruntfile.js]: This isn't needed - we use `jit-grunt` to load npm automatically.

File: package.json [hunk 2]
```diff
@@ -52,6 +53,7 @@
     "grunt-xmlmin": "^0.1.8",
     "jasmine-spec-reporter": "^4.2.1",
     "jit-grunt": "^0.10.0",
+    "jsdoc": "^3.6.2",
```
Comment (garethbowen) [inline package.json]: I don't think this is needed as it comes for free with `grunt-jsdoc`.

File: Gruntfile.js [hunk 3]
```diff
@@ -833,6 +833,14 @@ module.exports = function(grunt) {
       'build/ddocs/medic-admin/_attachments/js/templates.js':
         'build/ddocs/medic-admin/_attachments/js/templates.js',
     },
+    jsdoc : {
+      dist : {
+          src: ['webapp/src/js/*.js', 'admin/src/js/*.js'],
```
Comment (garethbowen) [inline Gruntfile.js]: When I run this the index.html is empty. I think this should be `['webapp/src/js/**/*.js', 'admin/src/js/**/*.js']` so it recurses into subdirectories.

**Comment by smbuthia** [comment]
Thanks @garethbowen. I've made the changes and also added the output folder to `.gitignore`.

**Pull request merged**